packages/core/src/codewhisperer/service/serviceContainer.ts

```diff
@@ -0,0 +1,33 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AuthUtil } from '../util/authUtil'
+import { ActiveStateController } from '../views/activeStateController'
+import { LineTracker } from '../tracker/lineTracker'
+
+export class Container {
+    static #instance: Container | undefined
+
+    static create(authProvider: AuthUtil): Container {
+        if (Container.#instance) {
+            throw new Error('Container already exists')
+        }
+
+        Container.#instance = new Container(authProvider)
+        return Container.#instance
+    }
+
+    static get instance(): Container {
+        return Container.#instance ?? Container.create(AuthUtil.instance)
```

Comment:
With the following you can get rid of the `create` method

```suggestion
        return Container.#instance ??= new Container(AuthUtil.instance)
```